tests/queries/test_query.py

```diff
@@ -158,3 +159,31 @@ def test_repr(self):
             repr(JoinPromoter("AND", 3, True)),
             "JoinPromoter(connector='AND', num_children=3, negated=True)",
         )
+
+
+class FromEmptyQuery(TestCase):
+    def test_boolean_annotation(self):
+        query = Query(None)
+        query.add_annotation(RawSQL("%s IS NULL", (None,), BooleanField()), "_check")
+        compiler = query.get_compiler(using="default")
+        result = compiler.execute_sql(SINGLE)
+        self.assertTrue(result[0])
```

Comment:
Use `assertIs(…, True/False)` for testing boolean values, rather than `assertTrue()` and `assertFalse()`, so you can check the actual boolean value, not the truthiness of the expression, as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).